llama-index-integrations/vector_stores/llama-index-vector-store-firestore/llama_index/vector_store/firestore/base.py

```diff
@@ -0,0 +1,218 @@
+"""Firestore vector store index.
+
+An index that is built on top of an existing vector store.
+
+"""
+
+import logging
+from typing import Any, List, Optional, Union
+
+import more_itertools
+from google.cloud.firestore import Client
+from google.cloud.firestore_v1 import And, FieldFilter, Or
+from google.cloud.firestore_v1.base_vector_query import DistanceMeasure
+from google.cloud.firestore_v1.document import DocumentSnapshot
+from google.cloud.firestore_v1.vector import Vector
+from llama_index.core.schema import BaseNode
+from llama_index.core.vector_stores.types import (
+    BasePydanticVectorStore,
+    ExactMatchFilter,
+    FilterCondition,
+    FilterOperator,
+    MetadataFilter,
+    MetadataFilters,
+    VectorStoreQuery,
+    VectorStoreQueryResult,
+)
+from llama_index.core.vector_stores.utils import (
+    metadata_dict_to_node,
+    node_to_metadata_dict,
+)
+
+DEFAULT_BATCH_SIZE = 500
+DEFAULT_TOP_K = 10
+
+_logger = logging.getLogger(__name__)
+
+
+def _to_firestore_operator(
+    operator: FilterOperator,
+) -> str:
+    """Convert from standard operator to Firestore operator."""
+    if operator == FilterOperator.EQ:
+        return "=="
+    if operator == FilterOperator.NE:
+        return "!="
+    if operator == FilterOperator.GT:
+        return ">"
+    if operator == FilterOperator.GTE:
+        return ">="
+    if operator == FilterOperator.LT:
+        return "<"
+    if operator == FilterOperator.LTE:
+        return "<="
+    if operator == FilterOperator.IN:
+        return "in"
+    if operator == FilterOperator.NIN:
+        return "not-in"
+    if operator == FilterOperator.CONTAINS:
+        return "array-contains"
+
+    raise ValueError(f"Operator {operator} not supported in Firestore.")
+
+
+def _to_firestore_filter(
+    standard_filters: MetadataFilters,
+) -> Union[FieldFilter, Or, And, List[FieldFilter]]:
+    """Convert from standard dataclass to filter dict."""
+    firestore_filters = []
+
+    # Convert standard filters to Firestore filters
+    # Firestore filters are FieldFilter or And/Or filters
+    for f in standard_filters.filters:
+        if isinstance(f, MetadataFilter):
+            firestore_filters.append(
+                FieldFilter(f.key, _to_firestore_operator(f.operator), f.value)
+            )
+        elif isinstance(f, ExactMatchFilter):
+            firestore_filters.append(FieldFilter(f.key, "==", f.value))
+
+    if len(firestore_filters) == 1:
+        return firestore_filters[0]
+    if standard_filters.condition == FilterCondition.AND:
+        return And(filters=firestore_filters)
+    if standard_filters.condition == FilterCondition.OR:
+        return Or(filters=firestore_filters)
+
+    return firestore_filters
```

Comment:
This can be an empty list [] in the code. 
do we have an unit test for that behavior? -- I didn't find the corresponding unit test for that.

Reply:
Modified to return None if list is empty, I will add test cases for filters.